### PR TITLE
Move the registration of the node from calicoctl into the container.

### PR DESCRIPTION
**Pull request by tomdee**
Also
- Clean up passing in felix env vars
- Print out the docker command that is used to run the container

### CODE REVIEW

**Comment by tomdee** [comment]
Let's talk design before I fix up the tests

**Comment by tomdee** [comment]
Discussed with Casey and plan is to move the checks and IP detection into the container. 

I'll raise an issue for improving the IP detection.

File: calico_node/Dockerfile [hunk 1]
```diff
@@ -11,15 +11,15 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM gliderlabs/alpine:edge
+FROM jeanblanchard/alpine-glibc
```
Comment (caseydavenport) [inline calico_node/Dockerfile]: What does this change give us?

File: calico_node/filesystem/etc/rc.local [hunk 2]
```diff
@@ -24,8 +26,10 @@ else
     # running twice ensures our starting configuration is correct.
```
Comment (caseydavenport) [inline calico_node/filesystem/etc/rc.local]: Now that we're printing logs here, so we want to still call it "policy-only mode"?  Just a thought per the discussions we had about renaming that.

File: calicoctl/calico_ctl/node.py [hunk 3]
```diff
@@ -419,23 +370,31 @@ def _start_node_container_docker(ip, ip6, log_dir, node_image, detach, etcd_envs
...
+        env_string += " -e " + an_env
 
-    if not detach:
-        _attach_and_stream(container)
+    vol_string = ""
+    for a_vol in binds:
+        vol_string += " -v %s:%s" % (a_vol, binds[a_vol]["bind"])
 
+    print "Running Docker container with the following command:\n"
+    print "docker run -d --restart=always --net=host --privileged --name=calico-node%s%s %s\n" % (env_string, vol_string, node_image)
```
Comment (caseydavenport) [inline calicoctl/calico_ctl/node.py]: Will this command be right if the `--detach=false` option is passed?  It hardcodes `-d`.

File: calicoctl/calico_ctl/node.py [hunk 4]
```diff
@@ -254,55 +254,9 @@ def node_start(node_image, runtime, log_dir, ip, ip6, as_num, detach,
...
-            print "No IP provided. Using detected IP: %s" % ip
-
-    # Verify that IPs are not already in use by another host.
-    error_if_bgp_ip_conflict(ip, ip6)
-
-    # Verify that the chosen IP exists on the current host
-    warn_if_unknown_ip(ip, ip6)
-
     # Warn if this hostname conflicts with an existing host
     warn_if_hostname_conflict(ip)
```
Comment (caseydavenport) [inline calicoctl/calico_ctl/node.py]: Why not move the hostname check?

File: calicoctl/calico_ctl/node.py [hunk 5]
```diff
@@ -316,53 +270,49 @@ def node_start(node_image, runtime, log_dir, ip, ip6, as_num, detach,
...
-    felix_envs = ["FELIX_ETCDADDR=%s" % etcd_authority,
-                  "FELIX_ETCDSCHEME=%s" % etcd_scheme]
 
     if etcd_ca_cert_file and etcd_key_file and etcd_cert_file:
         etcd_volumes.append(ETCD_CA_CERT_NODE_FILE)
         etcd_binds[etcd_ca_cert_file] = {"bind": ETCD_CA_CERT_NODE_FILE,
                                          "ro": True}
         etcd_envs.append("ETCD_CA_CERT_FILE=%s" % ETCD_CA_CERT_NODE_FILE)
-        felix_envs.append("FELIX_ETCDCAFILE=%s" % ETCD_CA_CERT_NODE_FILE)
+
```
Comment (caseydavenport) [inline calicoctl/calico_ctl/node.py]: Don't need the extra newlines here.

File: calicoctl/calico_ctl/node.py [hunk 6]
```diff
@@ -881,123 +783,20 @@ def handle_sigterm(sig, frame):
...
 
     output = docker_client.attach(container, stream=True)
     try:
         for raw_data in output:
             sys.stdout.write(raw_data)
+            if "Calico node started successfully" in raw_data:
+                stop_container_on_exit = False
+                break
     except KeyboardInterrupt:
         # mainline.  someone press Ctrl-C.
```
Comment (caseydavenport) [inline calicoctl/calico_ctl/node.py]: You didn't do this comment but it is yucky... What about: "**M**ainline.  **S**omeone press**ed** Ctrl-C"

File: calicoctl/startup.py [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+    Remove any existing IP address for this host's IPIP tunnel device.
+
+    Idempotent; does nothing if there is no IP assigned.  Releases the
+    IP from IPAM.
+    """
+    ip_addr = _get_host_tunnel_ip()
+    if ip_addr:
+        client.release_ips({ip_addr})
+    client.remove_per_host_config(hostname, "IpInIpTunnelAddr")
+
```
Comment (caseydavenport) [inline calicoctl/startup.py]: Another newline here.

File: calicoctl/startup.py [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+     # Get IP address of host, if none was specified
+    if not ip:
+        ips = get_host_ips(exclude=["^docker.*", "^cbr.*",
+                                    "virbr.*", "lxcbr.*", "veth.*",
+                                    "cali.*", "tunl.*"])
+        try:
+            ip = ips.pop()
+        except IndexError:
+            print "Couldn't autodetect a management IP address. Please provide" \
+                  " an IP by rerunning the command with the --ip=<IP_ADDRESS> flag."
```
Comment (caseydavenport) [inline calicoctl/startup.py]: Not sure this message makes sense now that it lives in the node container.

File: calico_node/Dockerfile (hunk 1 above)
Comment (tomdee) [inline calico_node/Dockerfile]: The ability to run a pyinstaller created startup script

**Comment by caseydavenport** [comment]
Once tests are working this LGTM.

**Pull request merged**